x-pack/plugins/cloud_security_posture/server/lib/check_for_findings.ts

```diff
@@ -6,15 +6,16 @@
  */
 
 import { ElasticsearchClient, type Logger } from '@kbn/core/server';
-import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../common/constants';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS, FINDINGS_INDEX_PATTERN } from '../../common/constants';
 
-export const isLatestFindingsIndexExists = async (
+export const checkForFindings = async (
   esClient: ElasticsearchClient,
+  latestIndex: boolean,
   logger: Logger
 ): Promise<boolean> => {
   try {
     const queryResult = await esClient.search({
-      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+      index: latestIndex ? LATEST_FINDINGS_INDEX_DEFAULT_NS : FINDINGS_INDEX_PATTERN,
```

Comment:
this change is a red flag for me, as far as I remember, this function is purposely checking for findings in the `latest-findings` index in order to return the `indexed` status which is used in the UI to display the pages and not the `no-findings` states.

This change will make it return true even when there are findings in general but not in the latest findings index, which is used by the UI pages to create the pages.